plasmapy/simulation/particle_tracker.py

```diff
@@ -0,0 +1,945 @@
+        of the particle and a time step based on the resolution of the grid. The candidate associated
+        with the gyroradius of the particle takes a ``time_steps_per_gyroperiod`` parameter that specifies
+        how many times the orbit of a gyrating particles will be subdivided. The other candidate,
+        associated with the spatial resolution of the grid object, calculates a time step using the time
+        it would take the fastest particle to cross some fraction of a grid cell length. This fraction is the Courant number.
+        """
+
+        if not self.is_adaptive_time_step:
+            raise ValueError(
+                "The setup adaptive time step method only applies to adaptive time steps!"
+            )
+
+        self._steps_per_gyroperiod = time_steps_per_gyroperiod
+        self._Courant_parameter = Courant_parameter
+
+    def _log(self, msg):
+        if self.verbose:
+            print(msg)  # noqa: T201
+
+    @particle_input
+    def load_particles(
+        self,
+        x,
+        v,
+        particle: Particle = Particle("p+"),  # noqa: B008
+    ):
+        r"""
+        Load arrays of particle positions and velocities.
+
+        Parameters
+        ----------
+        x : `~astropy.units.Quantity`, shape (N,3)
+            Positions for N particles
+
+        v : `~astropy.units.Quantity`, shape (N,3)
+            Velocities for N particles
+
+        particle : |particle-like|, optional
+            Representation of the particle species as either a |Particle| object
+            or a string representation. The default particle is protons.
+
+        distribution : str
+            A keyword which determines how particles will be distributed
+            in velocity space. Options are:
+
+                - 'monte-carlo': velocities will be chosen randomly,
+                    such that the flux per solid angle is uniform.
+
+                - 'uniform': velocities will be distributed such that,
+                   left unperturbed,they will form a uniform pattern
+                   on the detection plane.
+
+            Simulations run in the ``'uniform'`` mode will imprint a grid pattern
+            on the image, but will well-sample the field grid with a
+            smaller number of particles. The default is ``'monte-carlo'``.
+
+
+        """
+        # Raise an error if the run method has already been called.
+        self._enforce_order()
+
+        self.q = particle.charge.to(u.C).value
+        self.m = particle.mass.to(u.kg).value
+
+        if x.shape[0] != v.shape[0]:
+            raise ValueError(
+                "Provided x and v arrays have inconsistent numbers "
+                " of particles "
+                f"({x.shape[0]} and {v.shape[0]} respectively)."
+            )
+        else:
+            self.nparticles = x.shape[0]
+
+        self.x = x.to(u.m).value
+        self.v = v.to(u.m / u.s).value
+
+    def _stop_particles(self, particles_to_stop_mask):
```

Comment:
These methods need docstring - they can be a bit shorter (since they are private) but should still describe what the method does and define the parameters.